src/scripts/main.js

```diff
@@ -1 +1,80 @@
 'use strict';
+
+async function wait(ms) {
+  return new Promise((resolve) => setTimeout(resolve, ms));
+}
+
+const firstPromise = new Promise(async (resolve, reject) => {
+  let isTrue = false;
+
+  document.addEventListener('click', () => {
+    isTrue = true;
+    resolve('First promise was resolved');
+  });
+
+  await wait(3000);
+
+  if (!isTrue) {
+    reject(new Error('First promise was rejected'));
+  }
+});
+
+const secondPromise = new Promise((resolve, reject) => {
+  let isTrue = false;
+
+  document.addEventListener('click', () => {
+    resolve('Second promise was resolved');
+    isTrue = true;
+  });
+
+  if (isTrue !== true) {
+    document.addEventListener('contextmenu', () => {
+      resolve('Second promise was resolved');
+    });
+  }
+});
+
+const thirdPromise = new Promise((resolve, reject) => {
+  let firstClicked = false;
+  let twoClicked = false;
+
+  document.addEventListener('click', () => {
+    firstClicked = true;
+
+    if (twoClicked === true && firstClicked === true) {
+      resolve('Third promise was resolved');
+    }
+  });
+
+  document.addEventListener('contextmenu', (e) => {
+    twoClicked = true;
+    e.preventDefault();
+
+    if (twoClicked === true && firstClicked === true) {
+      resolve('Third promise was resolved');
+    }
+  });
+});
```

Comment:
In the `thirdPromise`, you're resolving the promise if both `firstClicked` and `twoClicked` are `true`. However, you're checking this condition in both event listeners. This means that the promise could be resolved twice, which is not allowed. A promise can only be settled (either resolved or rejected) once. All subsequent calls to resolve or reject are ignored. You need to ensure that your promise is only resolved once.